packages/xstate-wallet/src/workflows/approve-budget-and-fund.ts

```diff
@@ -110,12 +141,37 @@ export const approveBudgetAndFundWorkflow = (
     },
     sendResponse: (context: WorkflowContext, event) => {
       messagingService.sendResponse(context.requestId, serializeSiteBudget(context.budget));
-    }
+    },
+    sendBudgetUpdated: async (context: WorkflowContext, event) => {
+      await messagingService.sendBudgetNotification(context.budget);
+    },
+    updateBudgetToFree: assign(
+      (context: WorkflowContext): WorkflowContext => {
+        const clonedBudget = _.cloneDeep(context.budget);
+        // TODO: There must be a nicer way of mapping from one record to a new one
+        Object.keys(clonedBudget.budgets).forEach(k => {
+          clonedBudget.budgets[k] = freeAssetBudget(clonedBudget.budgets[k]);
+        });
+        return {...context, budget: clonedBudget};
```

Comment:
```suggestion
        const clonedBudget = _.mapValues(context.budget, freeAssetBudget);
        return {...context, budget: clonedBudget};
```

Reply:
I've updated it to something similar to your proposal